### ISSUE TITLE
Get more useful as a Plots backend

### ISSUE THREAD

**Issue by Evizero** [Evizero/UnicodePlots.jl#53]
@tbreloff Since I have to spend some time on UnicodePlots again soon to make it 0.5 ready anyway, I am considering what other task I should tackle while I am at it (aside from some open bugs). Do you have a list of suggestions that would be useful for Plots? For example, you mentioned the support for `ColorTypes` before, which is something that I am considering doing (I had to spend quite some time with those recently for a different project, so I am already a little familiar with the basic gist).

Anything else that comes to mind? Everyone is welcome too to chime in and propose things they would really like to see in UnicodePlots

**Comment by Evizero** [comment]
I got the impression that Plots is quite capable of doing it's own thing as long as it has a canvas available that offers some basic function. Am I correct in this @tbreloff ? I am also considering splitting UnicodePlots up into different packages.

- TerminalImageViewer.jl (to create image previews in the terminal. port of https://github.com/stefanhaustein/TerminalImageViewer)
- UnicodeCanvas.jl (low level drawing to basic canvas, no plot decorations)
- BaseShowTools.jl (very basic little plots to improve Base.show methods)
- UnicodePlots.jl (plot decorations and plotting API)

**Comment by tbreloff** [comment]
Yes basic support for drawing lines, polygons, and text is all you need.
